src/lib/dnssd/platform/tests/TestPlatform.cpp

```diff
@@ -152,102 +152,86 @@ test::ExpectedCall commissionableLargeEnhanced = test::ExpectedCall()
                                                      .AddSubtype("_V555")
                                                      .AddSubtype("_T70000")
                                                      .AddSubtype("_CM");
-void TestStub(nlTestSuite * inSuite, void * inContext)
+
+class TestDnssdPlatform : public ::testing::Test
+{
+public:
+    static void SetUpTestSuite() { VerifyOrDie(chip::Platform::MemoryInit() == CHIP_NO_ERROR); }
+    static void TearDownTestSuite()
+    {
+        DiscoveryImplPlatform::GetInstance().Shutdown();
```

Comment:
Yes, needs documentation explaining where the corresponding init is and why.

Reply:
I moved the common testcase initialization code into a method:
https://github.com/project-chip/connectedhomeip/blob/8c2c64e48cad20d16410ff57b16873c6a1b273d4/src/lib/dnssd/platform/tests/TestPlatform.cpp#L165-L171

Reply:
It is odd though that we have several inits (once per test run) and a single shutdown. A comment of why that is ok/expected may be useful here.